src/app/instance.h

```diff
@@ -76,23 +76,17 @@ class Instance {
   void InitClientContext();
 
   /// @brief Sets up (or reloads) client/server tunnels
+  /// @param is_reloading Are tunnels configuration in the process of reloading?
   /// @warning Configuration files must be parsed prior to setup
-  void SetupTunnels();
+  void SetupTunnels(bool is_reloading = false);
```

Comment:
>removing member m_IsReloading

1. Asumming that locks are not where they should be, if `Instance::Reload()` is called consecutively (multiple consecutive `SIGHUP`'s), I think we'll be looking at possible race conditions

2. `SetupTunnels()` with member `m_IsReloading` is a cheap way to resolve point 1 above

3. I find bool `is_reloading` as function arg to be somewhat confusing but I think the real issue is *how* we setup/reload tunnels.

Any other thoughts? How else could we approach these issues without more patching?